coil-video/src/main/java/coil/decode/VideoFrameDecoder.kt

```diff
@@ -172,7 +173,12 @@ class VideoFrameDecoder(
                 setDataSource("android.resource://${metadata.packageName}/${metadata.resId}")
             }
             else -> {
-                setDataSource(source.file().path)
+                // TODO check how to handle this cleanly
+                if (SDK_INT >= Build.VERSION_CODES.M) {
+                    setDataSource(source.mediaDataSource())
+                } else {
+                    setDataSource(source.file().toFile().path)
```

Comment:
Let's always use `setDataSource(source.file().toFile().path)` and avoid creating a custom media source (even if it's unsafe).